mbed-client-libservice/ip6string.h

```diff
@@ -18,14 +18,25 @@
 #ifdef __cplusplus
 extern "C" {
 #endif
+
+#include "ns_types.h"
 /**
  * Print binary IPv6 address to a string.
  * String must contain enough room for full address, 40 bytes exact.
  * IPv4 tunneling addresses are not covered.
  * \param ip6addr IPv6 address.
  * \p buffer to write string to.
  */
-void ip6tos(const void *ip6addr, char *p);
+uint8_t ip6tos(const void *ip6addr, char *p);
+
+/**
+ * Print binary IPv6 prefix to a string.
+ * String buffer must contain enough room for a full address and prefix length, 44 bytes exact.
+ * \param prefix IPv6 prefix.
+ * \param prefix_len length of prefix in bits.
+ * \p buffer to write string to.
+ */
+uint8_t ip6_prefix_tos(const uint8_t *prefix, uint8_t prefix_len, char *p);
```

Comment:
While touching this, fix the doxygen "\p" on both functions.

prefix being uint8_t * rather than void * is inconsistent with ip6tos.

Document what the behaviour is with extra bits beyond the prefix length - presumably they're not shown. And presumably extra bytes aren't read? So prefix_len 20 accesses 3 bytes, and prefix_len 0 accesses 0 bytes, making NULL, 0 legal.

Note that prefix length must be 0-128. Again, it count be uint_fast8_t.